index.js

```diff
@@ -204,11 +204,24 @@ function markdownTable(table, options) {
     rule = [];
 
     while (++index < cellCount) {
+      /* When `straighten` is false, make the rule the same size as the first row. */
+      if (settings.straighten === false) {
+        value = table[0][index];
+        if (value === null || value === undefined) {
```

Comment:
Sure, I actually copied this from a section further up - I should probably factor this out into a re-usable function TBH